### PR TITLE
GUI changes by st43225

### PR DESCRIPTION
**Pull request by bartic123**
- Dialogs for editing and deleting employees and bosses are now modal

- Added delete employee option

- Added alter employee option

- User now cannot enter id values for both employees and bosses, system is taking care of it itself therefore the id values are no longer accessible but only visible in dialogs

- Some unused imports were deleted

- Some other minor fixes in GUI

### CODE REVIEW

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 1]
```diff
@@ -9,22 +8,16 @@ public partial class EmployeeDialog : Form
         private MainForm main;
         private bool employee = false;
         private Boss bossForEmployee;
+        private Employee emp;
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: no abbreviations - attribute name should be employee

File: INPTP_AppForFixing/Entity/Boss.cs [hunk 2]
```diff
@@ -11,6 +9,7 @@ public class Boss : Employee
         private const int MONTHS_OF_YEAR = 12;
         private HashSet<Employee> employees;
         private Department department;
+        private const int FIRST_EMP_ID = 0;
```
Comment (rodi0878) [inline INPTP_AppForFixing/Entity/Boss.cs]: no abbreviations

File: INPTP_AppForFixing/MainForm.cs [hunk 3]
```diff
@@ -1,13 +1,20 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Windows.Forms;
 
 namespace INPTP_AppForFixing
 {
+    public enum Action
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: Action is name for library delegates. Use more descriptive name for this enum.

File: INPTP_AppForFixing/MainForm.cs [hunk 4]
```diff
@@ -44,18 +53,37 @@ public void OnEmployeeChange()
...
+                }
+            }
+        }
+
+        public Boss GetSelectedBoss() => (Boss)listBoxOfBosses.SelectedItem;
+
+        public Employee GetSelectedEmployee() => (Employee)listBoxEmpl.SelectedItem;
+
+
+        public int getNextBossId()
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: Boss is in fact an employee. Why bosses have separate id creation? IDs between bosses and employees should be same and unique.

File: INPTP_AppForFixing/MainForm.cs [hunk 5]
```diff
@@ -96,12 +124,44 @@ private void btnEmplAdd_Click(object sender, EventArgs e)
 
         private bool CheckSelectedBoss()
         {
-            return listBoxOfBosses.SelectedItem == null ? false : true;
+            return GetSelectedBoss() == null ? false : true;
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: unreadable ternary operator, better code: return GetSelectedBoss != null;

File: INPTP_AppForFixing/MainForm.cs [hunk 6]
```diff
@@ -96,12 +124,44 @@ private void btnEmplAdd_Click(object sender, EventArgs e)
...
+        }
+
+        private bool CheckSelectedEmp()
+        {
+            return GetSelectedEmployee() == null ? false : true;
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: same problem as previous method

File: INPTP_AppForFixing/MainForm.cs [hunk 7]
```diff
@@ -96,12 +124,44 @@ private void btnEmplAdd_Click(object sender, EventArgs e)
...
 
         private void listBoxOfBosses_SelectedIndexChanged(object sender, EventArgs e)
         {
             OnEmployeeChange();
         }
+
+        private void btnEmplDelete_Click(object sender, EventArgs e)
+        {
+            Boss boss = GetSelectedBoss();
+            Employee emp = GetSelectedEmployee();
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: no abbreviations

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 1 above)
Comment (bartic123) [reply INPTP_AppForFixing/EmployeeDialog.cs]: fixed

File: INPTP_AppForFixing/Entity/Boss.cs (hunk 2 above)
Comment (bartic123) [reply INPTP_AppForFixing/Entity/Boss.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 3 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 4 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 5 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 6 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 7 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs [hunk 8]
```diff
@@ -44,18 +53,37 @@ public void OnEmployeeChange()
...
+        public Boss GetSelectedBoss() => (Boss)listBoxOfBosses.SelectedItem;
+
+        public Employee GetSelectedEmployee() => (Employee)listBoxEmpl.SelectedItem;
+
+
+        public int getNextEmployeeId()
+        {
+            try
+            {
+                return Math.Max(bosses.Max(b => b.Id), bosses.Max(b => b.GetEmployees().Any() ? b.GetEmployees().Max(e => e.Id) : int.MinValue)) + 1;
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: Unhandled case - employee in boss in boss... And similair deeper nested cases

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 9]
```diff
@@ -1,58 +1,58 @@
...
-using System.Linq;
 using System.Windows.Forms;
 
 namespace INPTP_AppForFixing
 {
     public partial class EmployeeDialog : Form
     {
         private MainForm main;
-        private bool employee = false;
+        private bool isEmployee;
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: Names with verbs should be used for methods. For variables/attributes are preferred nouns.

File: INPTP_AppForFixing/MainForm.cs [hunk 10]
```diff
@@ -1,13 +1,20 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Windows.Forms;
 
 namespace INPTP_AppForFixing
 {
+    public enum Operation
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: public type should have own file

File: INPTP_AppForFixing/MainForm.cs [hunk 11]
```diff
@@ -96,12 +124,44 @@ private void btnEmplAdd_Click(object sender, EventArgs e)
 
         private bool CheckSelectedBoss()
         {
-            return listBoxOfBosses.SelectedItem == null ? false : true;
+            return GetSelectedBoss() != null;
+        }
+
+        private bool CheckSelectedEmp()
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: no abbreviations in names

File: INPTP_AppForFixing/MainForm.cs (hunk 8 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: What exactly do you mean by that?

Comment (rodi0878) [reply INPTP_AppForFixing/MainForm.cs]: You could have following structure:
Boss(1)
* Employee(2)
* Employee(3)
* Employee(4)
* Boss(5)
* * Employee(6)
* * Employee(7) 
* * Boss(8)
* * * Employee(9)
* Employee(10)

Current check wont check ids 6, 7, 8, 9.

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 9 above)
Comment (bartic123) [reply INPTP_AppForFixing/EmployeeDialog.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 10 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 8 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

File: INPTP_AppForFixing/MainForm.cs (hunk 11 above)
Comment (bartic123) [reply INPTP_AppForFixing/MainForm.cs]: fixed

**Pull request merged**